src/fsharp/TastOps.fs

```diff
@@ -3206,6 +3206,11 @@ let isSizeOfValRef g vref =
     // There is an internal version of typeof defined in prim-types.fs that needs to be detected
     || (g.compilingFslib && vref.LogicalName = "sizeof") 
 
+let isNameOfValRef g vref = 
+    valRefEq g vref g.nameof_vref 
+    // There is an internal version of typeof defined in prim-types.fs that needs to be detected
+    || (g.compilingFslib && vref.LogicalName = "nameof") 
```

Comment:
I see this pattern being used above, is it preferable to have a table of names that we check against rather than the raw string here?

Reply:
This is  orthogonal cleanup